File: Python/283.py
# Move Zeroes
# Easy

def moveZeroes(self, nums: 'list[int]') -> None:

    zcounter = 0
    insert_here = 0

    for i in range(0, len(nums)):
        if nums[i] == 0:
            zcounter += 1
            continue
        nums[insert_here] = nums[i]
        insert_here += 1

    for i in range(insert_here, len(nums)):
        nums[i] = 0

File: C++/283.cpp
// Move Zeroes
// Easy

#include <vector>

using namespace std;

void moveZeroes(vector<int>& nums) {
    
    int insert_here = 0;
    int z_counter   = 0;

    for ( int i = 0; i < nums.size(); i++ ) {
        if ( nums[i] == 0 ) {
            z_counter += 1;
            continue;
        }
        nums[insert_here] = nums[i];
        insert_here += 1;
    }

    for ( int i = insert_here; i < nums.size(); i++ )
        nums[i] = 0;
}